app/uk/gov/hmrc/tai/connectors/JourneyCacheConnector.scala

```diff
@@ -51,7 +51,9 @@ trait JourneyCacheConnector {
   }
 
   def cache(journeyName: String, data: Map[String,String])(implicit hc: HeaderCarrier): Future[Map[String,String]] = {
-    httpHandler.postToApi(cacheUrl(journeyName), data).map(_.json.as[Map[String,String]])
+    httpHandler.postToApi(
+      cacheUrl(journeyName), data
+    ).map(_.json.as[Map[String,String]])
```

Comment:
This is where we can make this API more flexible by allowing `cache` to take a type param here and that will specify the return type